### PR TITLE
Transcoding: muxers and demuxers

### PR DESCRIPTION
**Pull request by cameel**
This is pull request changes the transcoding task to use ffmpeg's `-f` option instead of file extension to select the **output** container.

Note that preservation of the **input** container type still relies on the file extension. That change still requires some work as discussed in https://github.com/golemfactory/golem/pull/4255#issuecomment-498333947. This pull request and https://github.com/golemfactory/ffmpeg-tools/pull/6 are just prerequisites for implementing it.

The code also updates `SUPPORTED_FILE_TYPES`, `SUPPORTED_VIDEO_CODECS` and `SUPPORTED_AUDIO_CODECS`. It looks like they're not used for anything so I'm not sure which formats should be there and which should not. I just added all of them - it already had nearly all of them anyway. Please tell me if that's not how it was supposed to be.

### Dependencies
This pull request is based on #4257 and should be merged after it.

For the code to work you also need a `golemfactory/ffmpeg-experimental` image built with https://github.com/golemfactory/ffmpeg-tools/pull/6. Tests in the CI won't pass until it's released on DockerHub.

**NOTE**: I have set the base branch of this pull request to `transcoding-remove-m3u-playlist-support-leftovers` (#4257) rather than `CGI/transcoding/master`.Please remember to change the base branch back to `CGI/transcoding/master` before you merge.

### CODE REVIEW

File: apps/transcoding/ffmpeg/task.py [hunk 1]
```diff
@@ -81,13 +83,30 @@ class ffmpegDefaults(TaskDefaults):
...
 
 class ffmpegTaskBuilder(TranscodingTaskBuilder):
-    SUPPORTED_FILE_TYPES = [Container.c_MKV, Container.c_AVI,
-                            Container.c_MP4, Container.c_MOV, Container.c_MPEG,
-                            Container.c_3GP, Container.c_3G2, Container.c_F4V]
-    SUPPORTED_VIDEO_CODECS = [VideoCodec.MPEG_1, VideoCodec.MPEG_2,
-                              VideoCodec.H_264, VideoCodec.H_265,
-                              VideoCodec.HEVC, VideoCodec.H_264]
-    SUPPORTED_AUDIO_CODECS = [AudioCodec.MP3, AudioCodec.AAC]
+    SUPPORTED_FILE_TYPES = [
```
Comment (nieznanysprawiciel) [inline apps/transcoding/ffmpeg/task.py]: These lists aren't used anymore as far as I see. Sorry I didn't remove them earlier. Can you do this ?

**Review by nieznanysprawiciel — APPROVED** [review]
Will be merged after fix to video extensions will be ready.

File: apps/transcoding/ffmpeg/task.py (hunk 1 above)
Comment (cameel) [reply apps/transcoding/ffmpeg/task.py]: Done.

**Comment by nieznanysprawiciel** [comment]
Merged in PR:
https://github.com/golemfactory/golem/pull/4407

**Pull request closed without merging**